### PR TITLE
Pal 880 update root documents for streaming

### PR DESCRIPTION
**Pull request by m78233**

### CODE REVIEW

**Review by developer01189998819991197253 — CHANGES_REQUESTED** [review]
reset changes to .png files, copy from root Palisade repo to fix formatting issues

File: README.md [hunk 1]
```diff
@@ -16,9 +16,43 @@ limitations under the License.
...
+This initial sequence of steps is completed with the Filtered-Resource Service prepared to provide the processed data.
+In the subsequent request by the client to the Data Service, the Filtered-Resource is requested to then proceed with passing this data to the client.
+The following diagram shows in a high level the services and their relative relationships with each other. 
+Communications between the services and their related support servers utilise Kafka streaming. 
+
+For a detailed description of the services' workflow see  
+<!--- 
+See doc/services.drawio for the source of this diagram
+--->
+[Appendix I Palisade Flow Diagram](#appendix-i-palisade-flow-diagram).
```
Comment (developer01189998819991197253) [inline README.md]: this link seems to be broken? where is it supposed to go?

File: README.md [hunk 2]
```diff
@@ -16,9 +16,43 @@ limitations under the License.
...
 # <img src="logos/logo.svg" width="180">
 
-## Scalable Data Access Policy Management and Enforcement
+### A Tool for Complex and Scalable Data Access Policy Enforcement
 
-### Project Build
+## Overview
+
+From the client’s perspective, they submit a request to examine data and receive a reference for this data. 
+This reference can then be used to view the data after it has been retrieved and possibly redacted or filtered based on the context of the query and the permissions for the user making the request.
```
Comment (developer01189998819991197253) [inline README.md]:
```suggestion
This reference can then be used to view the data after it has been retrieved and possibly redacted or filtered based on the context of the query and the permissions of the user making the request.
```

File: README.md [hunk 3]
```diff
@@ -16,9 +16,43 @@ limitations under the License.
...
+In the subsequent request by the client to the Data Service, the Filtered-Resource is requested to then proceed with passing this data to the client.
+The following diagram shows in a high level the services and their relative relationships with each other. 
+Communications between the services and their related support servers utilise Kafka streaming. 
+
+For a detailed description of the services' workflow see  
+<!--- 
+See doc/services.drawio for the source of this diagram
+--->
+[Appendix I Palisade Flow Diagram](#appendix-i-palisade-flow-diagram).  
+For a more technical description of each of the services, follow the related links below.
```
Comment (developer01189998819991197253) [inline README.md]:
```suggestion
For a more technical description of each of the services, follow the links below.
```

File: README.md [hunk 4]
```diff
@@ -16,9 +16,43 @@ limitations under the License.
...
+
+[Palisade Service](palisade-service/readme.md) takes client resource requests and starts the process.  
+[User Service](user-service/readme.md) identifies the user in the context of the Palisade service.  
+[Resource Service](resource-service/readme.md) identifies the resources that have been requested.  
+[Policy Service](policy-service/readme.md) provides the policy that are to be applied to the data.  
+[Attribute-Masking Service](attribute-masking-service/readme.md) creates a redacted version of the rescue.  
+[Topic-Offset Service](topic-offset-service/readme.md) identifies the start and end of the resources for a request.  
+[Filtered-Resource Service](filtered-resource-service/readme.md) readies the resources and later handles passing the data to the client.  
+[Data Service](data-service/readme.md) initiates the process of passing the resources to the client.  
+[Audit Service](audit-service/readme.md) operates in the background providing an audit record of the process.
```
Comment (developer01189998819991197253) [inline README.md]: Change this to be in alphabetical order

**Review by developer01189998819991197253 — CHANGES_REQUESTED** [review]
Read through the rest of the root readme, update capitalisations, add Kafka info. 
fix png changes

File: README.md (hunk 1 above)
Comment (m78233) [reply README.md]: It is a link to the appendix at the end of this page where I have put the palisade flow diagram.

File: README.md (hunk 4 above)
Comment (m78233) [reply README.md]: done

File: README.md [hunk 5]
```diff
@@ -16,9 +16,39 @@ limitations under the License.
 
 # <img src="logos/logo.svg" width="180">
 
-## Scalable Data Access Policy Management and Enforcement
+### A Tool for Complex and Scalable Data Access Policy Enforcement
```
Comment (developer01189998819991197253) [inline README.md]:
```suggestion
## A Tool for Complex and Scalable Data Access Policy Enforcement
```

File: README.md [hunk 6]
```diff
@@ -16,9 +16,39 @@ limitations under the License.
...
+
+For a detailed description of the services' workflow see [Appendix I - Palisade Flow Diagram](#appendix-i).  
+For a more technical description of each of the services, follow the links below.
+
+<!--- 
+See doc/services.drawio for the source of this diagram
+--->
+![Service Sequence diagram](doc/services.png)
+
+[Attribute-Masking Service](attribute-masking-service/README.md) creates a redacted version of the rescue.
```
Comment (developer01189998819991197253) [inline README.md]: redacted version of the rescue?

File: README.md [hunk 7]
```diff
@@ -16,9 +16,39 @@ limitations under the License.
...
+For a detailed description of the services' workflow see [Appendix I - Palisade Flow Diagram](#appendix-i).  
+For a more technical description of each of the services, follow the links below.
+
+<!--- 
+See doc/services.drawio for the source of this diagram
+--->
+![Service Sequence diagram](doc/services.png)
+
+[Attribute-Masking Service](attribute-masking-service/README.md) creates a redacted version of the rescue.  
+[Audit Service](audit-service/README.md) operates in the background providing an audit record of the process.
```
Comment (developer01189998819991197253) [inline README.md]:
```suggestion
[Audit Service](audit-service/README.md) used for auditing successful and failed requests for data in Palisade.  
```

File: README.md [hunk 8]
```diff
@@ -16,9 +16,39 @@ limitations under the License.
...
-## Scalable Data Access Policy Management and Enforcement
+### A Tool for Complex and Scalable Data Access Policy Enforcement
 
-### Project Build
+## Overview
+
+From the client’s perspective, they submit a request to examine data and receive a reference for this data. 
+This reference can then be used to view the data after it has been retrieved and possibly redacted or filtered based on the context of the query and the permissions of the user making the request.
+
+Under the hood, the request involves sending a message that is processed sequentially by a set of the micro-services starting first with Palisade Service.
```
Comment (dev930018) [inline README.md]: A note after this saying something like "This service returns a token, the reference to be used for the rest of the request"

File: README.md [hunk 9]
```diff
@@ -16,9 +16,39 @@ limitations under the License.
...
+For a more technical description of each of the services, follow the links below.
+
+<!--- 
+See doc/services.drawio for the source of this diagram
+--->
+![Service Sequence diagram](doc/services.png)
+
+[Attribute-Masking Service](attribute-masking-service/README.md) creates a redacted version of the rescue.  
+[Audit Service](audit-service/README.md) operates in the background providing an audit record of the process.  
+[Data Service](data-service/README.md) initiates the process of passing the resources to the client.
```
Comment (developer01189998819991197253) [inline README.md]: doesnt sound accurate. the Data Service is used to connect to actual backing stores containing data to returned processed resources

File: README.md [hunk 10]
```diff
@@ -16,9 +16,39 @@ limitations under the License.
...
+
+<!--- 
+See doc/services.drawio for the source of this diagram
+--->
+![Service Sequence diagram](doc/services.png)
+
+[Attribute-Masking Service](attribute-masking-service/README.md) creates a redacted version of the rescue.  
+[Audit Service](audit-service/README.md) operates in the background providing an audit record of the process.  
+[Data Service](data-service/README.md) initiates the process of passing the resources to the client.  
+[Filtered-Resource Service](filtered-resource-service/README.md) readies the resources and later handles passing the data to the client.
```
Comment (developer01189998819991197253) [inline README.md:52]: again doesnt sound totally accurate

File: README.md [hunk 11]
```diff
@@ -16,9 +16,39 @@ limitations under the License.
...
 
-### Project Build
+## Overview
+
+From the client’s perspective, they submit a request to examine data and receive a reference for this data. 
+This reference can then be used to view the data after it has been retrieved and possibly redacted or filtered based on the context of the query and the permissions of the user making the request.
+
+Under the hood, the request involves sending a message that is processed sequentially by a set of the micro-services starting first with Palisade Service.
+With each service, the message is modified and then forwarded onto the next service with the end goal of producing resources ready to be viewed by the client.
+This initial sequence of steps is completed with the Filtered-Resource Service prepared to provide the processed data.
```
Comment (dev930018) [inline README.md]: Another note, "This service returns, one-by-one upon request over a websocket, metadata for a single resource (e.g. a file) they are authorised to view in some capacity."

File: README.md [hunk 12]
```diff
@@ -16,9 +16,39 @@ limitations under the License.
...
+In the subsequent request by the client to the Data Service, the Filtered-Resource is requested to then proceed with passing this data to the client.
```
Comment (dev930018) [inline README.md]: This doesn't read too good, maybe something like
```suggestion
In the subsequent request by the client to the Data Service, the service reads the resource and returns it to the client (after applying relevant policies to the data).
```
?

File: README.md [hunk 13]
```diff
@@ -16,9 +16,39 @@ limitations under the License.
...
+The following diagram shows in a high level the services and their relative relationships with each other.
```
Comment (dev930018) [inline README.md]:
```suggestion
The following diagram shows at a high level the services and their relative relationships with each other. 
```

File: README.md [hunk 14]
```diff
@@ -16,9 +16,39 @@ limitations under the License.
...
+
+[Attribute-Masking Service](attribute-masking-service/README.md) creates a redacted version of the rescue.  
+[Audit Service](audit-service/README.md) operates in the background providing an audit record of the process.  
+[Data Service](data-service/README.md) initiates the process of passing the resources to the client.  
+[Filtered-Resource Service](filtered-resource-service/README.md) readies the resources and later handles passing the data to the client.  
+[Palisade Service](palisade-service/README.md) takes client resource requests and starts the process.  
+[Policy Service](policy-service/README.md) provides the policy that are to be applied to the data.  
+[Resource Service](resource-service/README.md) identifies the resources that have been requested.  
+[Topic-Offset Service](topic-offset-service/README.md) identifies the start and end of the resources for a request.  
+[User Service](user-service/README.md) identifies the user in the context of the Palisade service.
```
Comment (dev930018) [inline README.md]: Order these as they process messages, not alphabetically:
* palisade
* user
* resource
* attribute-masking
* topic-offset
* filtered-resource

_leave a gap_

* data

_leave a gap_

* audit

File: README.md [hunk 15]
```diff
@@ -16,9 +16,39 @@ limitations under the License.
...
+<!--- 
+See doc/services.drawio for the source of this diagram
+--->
+![Service Sequence diagram](doc/services.png)
+
+[Attribute-Masking Service](attribute-masking-service/README.md) creates a redacted version of the rescue.  
+[Audit Service](audit-service/README.md) operates in the background providing an audit record of the process.  
+[Data Service](data-service/README.md) initiates the process of passing the resources to the client.  
+[Filtered-Resource Service](filtered-resource-service/README.md) readies the resources and later handles passing the data to the client.  
+[Palisade Service](palisade-service/README.md) takes client resource requests and starts the process.
```
Comment (developer01189998819991197253) [inline README.md]:
```suggestion
[Palisade Service](palisade-service/README.md) recieves a clients request, returns a unique token and initiates the processing of resources.
```

File: README.md [hunk 16]
```diff
@@ -16,9 +16,39 @@ limitations under the License.
...
+See doc/services.drawio for the source of this diagram
+--->
+![Service Sequence diagram](doc/services.png)
+
+[Attribute-Masking Service](attribute-masking-service/README.md) creates a redacted version of the rescue.  
+[Audit Service](audit-service/README.md) operates in the background providing an audit record of the process.  
+[Data Service](data-service/README.md) initiates the process of passing the resources to the client.  
+[Filtered-Resource Service](filtered-resource-service/README.md) readies the resources and later handles passing the data to the client.  
+[Palisade Service](palisade-service/README.md) takes client resource requests and starts the process.  
+[Policy Service](policy-service/README.md) provides the policy that are to be applied to the data.
```
Comment (developer01189998819991197253) [inline README.md]:
```suggestion
[Policy Service](policy-service/README.md) provides the policies that apply to the resources.
```

File: attribute-masking-service/README.md [hunk 17]
```diff
@@ -16,7 +16,7 @@ limitations under the License.
 
 # <img src="../logos/logo.svg" width="180">
 
-# Palisade Attribute-Masking-Service
+# Attribute-Masking-Service
```
Comment (dev930018) [inline attribute-masking-service/README.md]:
```suggestion
# Attribute-Masking Service
```

**Review by dev930018 — CHANGES_REQUESTED** [review]

File: README.md [hunk 18]
```diff
@@ -16,9 +16,39 @@ limitations under the License.
...
+--->
+![Service Sequence diagram](doc/services.png)
+
+[Attribute-Masking Service](attribute-masking-service/README.md) creates a redacted version of the rescue.  
+[Audit Service](audit-service/README.md) operates in the background providing an audit record of the process.  
+[Data Service](data-service/README.md) initiates the process of passing the resources to the client.  
+[Filtered-Resource Service](filtered-resource-service/README.md) readies the resources and later handles passing the data to the client.  
+[Palisade Service](palisade-service/README.md) takes client resource requests and starts the process.  
+[Policy Service](policy-service/README.md) provides the policy that are to be applied to the data.  
+[Resource Service](resource-service/README.md) identifies the resources that have been requested.
```
Comment (developer01189998819991197253) [inline README.md]:
```suggestion
[Resource Service](resource-service/README.md) communicates with the backing stores to ensure the resource exists.
```

File: README.md [hunk 19]
```diff
@@ -16,9 +16,39 @@ limitations under the License.
...
+The following diagram shows in a high level the services and their relative relationships with each other. 
+Communications between the services and their related support servers utilise Kafka streaming. 
+
+For a detailed description of the services' workflow see [Appendix I - Palisade Flow Diagram](#appendix-i).  
+For a more technical description of each of the services, follow the links below.
+
+<!--- 
+See doc/services.drawio for the source of this diagram
+--->
+![Service Sequence diagram](doc/services.png)
```
Comment (dev930018) [inline README.md:44]: Data-Svc goes to AMS, not FRS
TOS goes from AMS to FRS as current, but AMS also goes straight to FRS (like a triangle shape between the three)
Audit Svc lines could be tidier

**Review by dev930018 — CHANGES_REQUESTED** [review]

File: README.md [hunk 20]
```diff
@@ -16,9 +16,39 @@ limitations under the License.
...
+
+[Attribute-Masking Service](attribute-masking-service/README.md) creates a redacted version of the rescue.  
+[Audit Service](audit-service/README.md) operates in the background providing an audit record of the process.  
+[Data Service](data-service/README.md) initiates the process of passing the resources to the client.  
+[Filtered-Resource Service](filtered-resource-service/README.md) readies the resources and later handles passing the data to the client.  
+[Palisade Service](palisade-service/README.md) takes client resource requests and starts the process.  
+[Policy Service](policy-service/README.md) provides the policy that are to be applied to the data.  
+[Resource Service](resource-service/README.md) identifies the resources that have been requested.  
+[Topic-Offset Service](topic-offset-service/README.md) identifies the start and end of the resources for a request.
```
Comment (developer01189998819991197253) [inline README.md]: something like this sounds better
```
[Topic-Offset Service](topic-offset-service/README.md) listens to kafka topics to inform the Filtered-Resource Service on when messages are ready to be processed.
```

File: README.md (hunk 14 above)
Comment (developer01189998819991197253) [inline README.md]:
```suggestion
[User Service](user-service/README.md) connects to User Service providers to ensure the user exists.  
```

File: README.md (hunk 8 above)
Comment (m78233) [reply README.md]: Have tied the idea of a reference and a token in the description.

File: README.md (hunk 11 above)
Comment (m78233) [reply README.md]: Had a go at weaving this into the sentence "In the subsequent request by the client ...."  .

File: README.md (hunk 14 above)
Comment (m78233) [reply README.md]: done

File: README.md (hunk 19 above)
Comment (m78233) [reply README.md:44]: Ok had another go at it.

File: README.md [hunk 21]
```diff
@@ -16,9 +16,45 @@ limitations under the License.
...
+
+<!--- 
+See doc/services.drawio for the source of this diagram
+--->
+![Service Sequence diagram](doc/services.png)
+
+[Palisade Service](palisade-service/README.md) takes client resource requests and starts the process.  
+[User Service](user-service/README.md) identifies the user in the context of the Palisade service.  
+[Resource Service](resource-service/README.md) identifies the resources that have been requested.  
+[Policy Service](policy-service/README.md) provides the policy that are to be applied to the data.
```
Comment (developer01189998819991197253) [inline README.md]:
```suggestion
[Policy Service](policy-service/README.md) checks that policies exist for each resource in a request.  
```

File: README.md [hunk 22]
```diff
@@ -16,9 +16,45 @@ limitations under the License.
...
+<!--- 
+See doc/services.drawio for the source of this diagram
+--->
+![Service Sequence diagram](doc/services.png)
+
+[Palisade Service](palisade-service/README.md) takes client resource requests and starts the process.  
+[User Service](user-service/README.md) identifies the user in the context of the Palisade service.  
+[Resource Service](resource-service/README.md) identifies the resources that have been requested.  
+[Policy Service](policy-service/README.md) provides the policy that are to be applied to the data.  
+[Attribute-Masking Service](attribute-masking-service/README.md) creates a redacted version of the rescue.
```
Comment (developer01189998819991197253) [inline README.md]:
```suggestion
[Attribute-Masking Service](attribute-masking-service/README.md) applies policies from the Policy Service against the resources in the request.
```

File: README.md [hunk 23]
```diff
@@ -16,9 +16,45 @@ limitations under the License.
...
+
+[Palisade Service](palisade-service/README.md) takes client resource requests and starts the process.  
+[User Service](user-service/README.md) identifies the user in the context of the Palisade service.  
+[Resource Service](resource-service/README.md) identifies the resources that have been requested.  
+[Policy Service](policy-service/README.md) provides the policy that are to be applied to the data.  
+[Attribute-Masking Service](attribute-masking-service/README.md) creates a redacted version of the rescue.  
+[Topic-Offset Service](topic-offset-service/README.md) identifies the start and end of the resources for a request.  
+[Filtered-Resource Service](filtered-resource-service/README.md) readies the resources and later handles passing the data to the client.  
+
+[Data Service](data-service/README.md) initiates the process of passing the resources to the client.
```
Comment (developer01189998819991197253) [inline README.md]:
```suggestion
[Data Service](data-service/README.md) retrieves resources from the relevant backing stores. 
```

File: README.md [hunk 24]
```diff
@@ -16,9 +16,45 @@ limitations under the License.
...
+[User Service](user-service/README.md) identifies the user in the context of the Palisade service.  
+[Resource Service](resource-service/README.md) identifies the resources that have been requested.  
+[Policy Service](policy-service/README.md) provides the policy that are to be applied to the data.  
+[Attribute-Masking Service](attribute-masking-service/README.md) creates a redacted version of the rescue.  
+[Topic-Offset Service](topic-offset-service/README.md) identifies the start and end of the resources for a request.  
+[Filtered-Resource Service](filtered-resource-service/README.md) readies the resources and later handles passing the data to the client.  
+
+[Data Service](data-service/README.md) initiates the process of passing the resources to the client.  
+
+[Audit Service](audit-service/README.md) operates in the background providing an audit record of the process.
```
Comment (developer01189998819991197253) [inline README.md]:
```suggestion
[Audit Service](audit-service/README.md) operates in the background providing an audit log for each request.    
```

File: README.md [hunk 25]
```diff
@@ -16,9 +16,45 @@ limitations under the License.
...
+For a detailed description of the services' workflow see [Appendix I - Palisade Flow Diagram](#appendix-i).  
+For a more technical description of each of the services, follow the links below.
+
+<!--- 
+See doc/services.drawio for the source of this diagram
+--->
+![Service Sequence diagram](doc/services.png)
+
+[Palisade Service](palisade-service/README.md) takes client resource requests and starts the process.  
+[User Service](user-service/README.md) identifies the user in the context of the Palisade service.
```
Comment (developer01189998819991197253) [inline README.md]: Although worded not quite right, something like below is more descriptive of the User Service
```
[User Service](user-service/README.md) ensures that the user is valid by checking they exist in a user provider technology.
```

**Review by developer01189998819991197253 — CHANGES_REQUESTED** [review]

File: README.md (hunk 25 above)
Comment (m78233) [reply README.md]: Went with the first suggestion for user service

File: README.md [hunk 26]
```diff
@@ -16,9 +16,45 @@ limitations under the License.
...
-## Scalable Data Access Policy Management and Enforcement
+## A Tool for Complex and Scalable Data Access Policy Enforcement
 
-### Project Build
+## Overview
+
+From the client’s perspective, they submit a request to examine data and receive a reference for this data. 
+This reference can then be used to view the data after it has been retrieved and possibly redacted or filtered based on the context of the query and the permissions of the user making the request.
+
+Under the hood, the request involves sending a message that is processed sequentially by a set of the micro-services starting first with Palisade Service.
```
Comment (dev930018) [inline README.md]:
```suggestion
Under the hood, the request involves sending a message that is processed sequentially by a set of micro-services starting first with Palisade Service.
```

File: README.md [hunk 27]
```diff
@@ -16,9 +16,45 @@ limitations under the License.
...
+The Palisade Service will perform two tasks.
+First it will return a reference, a token back to the client that will uniquely identify this request.
+This token is used throughout the application to tie every aspect of the process to this unique identifier and is later used by the client to retrieve the resources available for this request. 
+The second task for the Palisade Service is to initiate the processing of the request by forwarding it onto the next service in the sequence (in this case the User Service).
+The service will perform its required task and then forward the message onto the next service with the end goal of producing resources ready to be viewed by the client.
+This sequence of steps is completed with the Filtered-Resource Service prepared to provide the processed data.
+In the subsequent request by the client to the Data Service, the service returns, one-by-one upon request over a websocket, metadata for a single resource (e.g. a file) they are authorised to view in some capacity.
```
Comment (dev930018) [inline README.md]:
```suggestion
In the subsequent request by the client to the Filtered-Resource Service, the service returns, one-by-one upon request over a websocket, metadata for a single resource (e.g. a file) they are authorised to view in some capacity.
These resources can then be read from the Data Service.
```

File: README.md (hunk 20 above)
Comment (dev930018) [reply README.md]:
```suggestion
[Topic-Offset Service](topic-offset-service/README.md) listens to kafka topics to inform the Filtered-Resource Service where on the topic (at what offset) the first message for each token appears.
```

**Review by dev930018 — CHANGES_REQUESTED** [review]

**Review by dev930018 — APPROVED** [review]

**Pull request merged**